mesonbuild/modules/qt.py

```diff
@@ -196,15 +204,43 @@ def preprocess(self, state, args, kwargs):
         return ModuleReturnValue(sources, sources)
 
     @FeatureNew('qt.compile_translations', '0.44.0')
-    @permittedKwargs({'ts_files', 'install', 'install_dir', 'build_by_default', 'method'})
+    @FeatureNewKwargs('qt.compile_translations', '0.56.0', ['qresource'])
+    @FeatureNewKwargs('qt.compile_translations', '0.56.0', ['rcc_extra_arguments'])
+    @permittedKwargs({'ts_files', 'qresource', 'rcc_extra_arguments', 'install', 'install_dir', 'build_by_default', 'method'})
     def compile_translations(self, state, args, kwargs):
-        ts_files, install_dir = [extract_as_list(kwargs, c, pop=True) for c in  ['ts_files', 'install_dir']]
+        ts_files, install_dir = [extract_as_list(kwargs, c, pop=True) for c in ['ts_files', 'install_dir']]
+        qresource = kwargs.get('qresource')
+        if qresource:
+            if ts_files:
+                raise MesonException('qt.compile_translations: Cannot specify both ts_files and qresource')
+            if os.path.dirname(qresource) != '':
+                raise MesonException('qt.compile_translations: qresource file name must not contain a subdirectory.')
+            qresource = File.from_built_file(state.subdir, qresource)
+            infile_abs = os.path.join(state.environment.source_dir, qresource.relative_name())
+            outfile_abs = os.path.join(state.environment.build_dir, qresource.relative_name())
+            os.makedirs(os.path.dirname(outfile_abs), exist_ok=True)
+            shutil.copyfile(infile_abs, outfile_abs)
+            shutil.copystat(infile_abs, outfile_abs)
```

Comment:
couldn't you just use `shutil.copy2(infile_abs, outfile_abs)` here?

Reply:
I could, the code I copied it from used this method though.

I'll take a look at reviewing meson's use of copyfile and copystat rather than copy2, while I'm at it.

Reply:
https://github.com/mesonbuild/meson/pull/7619